### PR TITLE
Update Emscripten to 3.1.18

### PR DESCRIPTION
**Pull request by hoodmane**
As requested by @tiran. Maybe we'll get lucky and opencv-python and xgboost will build on this version...

### CODE REVIEW

**Comment by hoodmane** [comment]
CF https://github.com/pyodide/pyodide/issues/2951

File: Makefile.envs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 export PYVERSION ?= 3.10.2
 # Note: when updating EMSCRIPTEN_VERSION make sure to update
 # the version number in the "uname" patch.
-export PYODIDE_EMSCRIPTEN_VERSION ?= 3.1.14
+export PYODIDE_EMSCRIPTEN_VERSION ?= 3.1.15
```
Comment (tiran) [inline Makefile.envs:4]:
```suggestion
export PYODIDE_EMSCRIPTEN_VERSION ?= 3.1.18
```

Comment (hoodmane) [reply Makefile.envs:4]: Oops...

**Comment by ryanking13** [comment]
I still don't understand what [the log](https://github.com/pyodide/pyodide/runs/7725790795?check_suite_focus=true#step:9:108)

> 0 is not a valid OpCode

means, and how it is (possibly) releated to the test failures.

**Comment by hoodmane** [comment]
Me neither. But at least xgboost and opencv build here...

**Comment by hoodmane** [comment]
So when we tried to update to 3.1.15, it broke building of xgboost and opencv. Trying to update to 3.1.18 breaks building of RobotRaconteur and testing of pyproj. Though I actually have a theory for why the RobotRaconteur build is broken.

**Comment by ryanking13** [comment]
> So when we tried to update to 3.1.15, it broke building of xgboost and opencv.

In 3.1.15, xgboost and opencv builds, but it was the test that failed.

**Comment by ryanking13** [comment]
I did some binary search among commits in `emscripten-releases` which emsdk uses, and the error happend after this commit: https://chromium.googlesource.com/emscripten-releases/+/c3f55f5c179c073c569684a0640727edb7c5d10e.

However, that commit itself includes 24 commits of `llvm-project`... (sigh)

**Comment by hoodmane** [comment]
Now failing with:
```
Error: bad export type for `gk_cur_jbufs`: undefined
```

**Comment by ryanking13** [comment]
> Now `scipy.linalg` is failing with:
> 
> ```
> Error: bad export type for `gk_cur_jbufs`: undefined
> ```

 `suitesparse` seems failing.

**Comment by ryanking13** [comment]
```
[LinkError: WebAssembly.instantiate(): Import #77 module="env" function="mktime" error: imported function does not match the expected type]
```
It seems like the error is related to the 64bit `time_t` update and hopefully correctly passing LDFLAGS would fix it.

**Comment by ryanking13** [comment]
It looks like it is building now except for minor test errors.

**Review by ryanking13 — APPROVED** [review]
Thanks a lot @hoodmane! Please update the changelog.

Probably we can fix `RobotRaconteur` failure after merging this.

**Comment by hoodmane** [comment]
Even on the main branch, `RobotRaconteur` build fails locally for me.

**Pull request merged**